tests/test_syntax/test_inline_comb.py

```diff
@@ -136,3 +137,385 @@ def logic():
     tester.compile_and_run("verilator", skip_compile=True,
                            directory=os.path.join(os.path.dirname(__file__),
                                                   "build"))
+
+
+def test_csr_comb():
```

Comment:
Not disputing that it is so, but unclear to me why a simpler test example is difficult to construct? If my understanding is correct the issue was triggered when there were more than 10 new variables -- can this be deterministically hit in the case of 10 or more `elif`s? Or 10 or more separate variables?

If nothing else, would be good to change the name of the test to something more representative of the error, ala `test_many_name_gen` and add a comment at top of this function saying like "tests for issue ____ triggered when ____. actual circuit contents don't matter, just need complex circuit to trigger condition"

Reply:
Actually I think it needs 11 separate variables and 11 separate elifs.  In this situations, we could get a name with 111 which could be case number 11 and variable 1 or case number 1 and variable number 11.

I can try to create a contrived example that exercises the problem, I think I originally just thought that the contrived example would have to be so large/complex that there wasn't much value in creating it versus just using the code that was already written that produced the problem.

Reply:
Confirmed it's triggered with 11 variables and 11 elifs (because ssa appends a unique ID for every variable, so we'll get the 111 ambiguous case above without the "_" ending the prefix).  I added a test case that just has 11 variables and 11 cases that assign various values and triggers the type error induced by this problem.